test/util/generator_helpers.rb

```diff
@@ -146,5 +144,15 @@ def generate
                        fixture_info['options'],
                        fixture_info['spec'])
     end
+
+    # Validates multi-platform specs
+    get_platforms(0).zip(
```

Comment:
More deterministic would definitely be better...as it is the combination of `get_platforms` and `simple_roles.cycle` already leads to a level of indeterminism (even before this commit in the origingal fixture generation code) in the sense that insertions (rather than appends) to the `osinfo` data structure leads to different combinations of platforms and simple roles.